Master branch fails tests.

```bash
$ npm test

# should register a default path
/workspace/server-router/node_modules/wayfarer/index.js:69
    throw new Error("route '" + route + "' did not match")
    ^

Error: route 'GET/bar' did not match
    at match (/workspace/server-router/node_modules/wayfarer/index.js:69:11)
    at ServerRouter.emit [as _router] (/workspace/server-router/node_modules/wayfarer/index.js:49:19)
    at ServerRouter.match (/workspace/server-router/index.js:40:15)
    at Server.handler (/workspace/server-router/test.js:56:14)
    at Server.emit (events.js:189:13)
    at parserOnIncoming (_http_server.js:676:12)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:109:17)
npm ERR! Test failed.  See above for more details.
```

--- a/README.md
+++ b/README.md
@@ -10,14 +10,19 @@ streaming servers.
 var serverRouter = require('server-router')
 var http = require('http')
 
-var router = serverRouter()
+var router = serverRouter({ default: '/404' })
 
-router.route('GET', '/hello', function (req, res, ctx) {
+router.route('GET', '/hello', function (req, res, params) {
   res.end('hello world')
 })
 
 router.route('PUT', '/hello/:name', function (req, res, params) {
   res.end('hi there ' + params.name)
+})
+
+router.route('', '/404', function (req, res, params) {
+  res.status = 404
+  res.end('404')
 })
 
 http.createServer(router.start()).listen()
@@ -27,7 +32,7 @@ http.createServer(router.start()).listen()
 ### router = serverRouter(opts)
 Create a new router with opts.
 
-### router.route(method|routes, route, function(req, res, params))
+### router.route(method|methods, route, function (req, res, params))
 Register a new route with an HTTP method name and a routename. Can register
 multiple handlers by passing an array of method names. `params` contains
 matched partials from the route.

--- a/README.md
+++ b/README.md
@@ -32,7 +32,7 @@ http.createServer(router.start()).listen()
 ### router = serverRouter(opts)
 Create a new router with opts.
 
-### router.route(method|methods, route, function (req, res, params))
+### router.route(method|[methods], route, function (req, res, params))
 Register a new route with an HTTP method name and a routename. Can register
 multiple handlers by passing an array of method names. `params` contains
 matched partials from the route.

--- a/index.js
+++ b/index.js
@@ -9,29 +9,36 @@ function ServerRouter (opts) {
   if (!(this instanceof ServerRouter)) return new ServerRouter(opts)
 
   opts = opts || {}
-  assert.equal(typeof opts, 'object', 'server-router: opts should be type object')
-  this._router = wayfarer()
+  assert.strictEqual(typeof opts, 'object', 'server-router: opts should be type object')
+  this._router = wayfarer(opts.default)
+  this._default = opts.default
 }
 
 ServerRouter.prototype.route = function (method, route, handler) {
   assert.ok(typeof method === 'string' || Array.isArray(method), 'server-router.route: method should be type string or array')
-  assert.equal(typeof route, 'string', 'server-router.route: route should be type string')
-  assert.equal(typeof handler, 'function', 'server-router.route: handler should be type function')
+  assert.strictEqual(typeof route, 'string', 'server-router.route: route should be type string')
+  assert.strictEqual(typeof handler, 'function', 'server-router.route: handler should be type function')
 
   if (Array.isArray(method)) {
     var methodRoute = null
     for (var i = 0; i < method.length; i++) {
-      methodRoute = method[i] + '/' + route.replace(/^[#/]/, '')
+      methodRoute = method[i].toUpperCase() + '/' + route.replace(/^[#/]/, '')
       this._router.on(methodRoute, function (params, req, res) {
         handler(req, res, params)
       })
     }
-  } else {
-    route = method.toUpperCase() + '/' + route.replace(/^[#/]/, '')
+    return
+  }
+  if (this._default === route) {
     this._router.on(route, function (params, req, res) {
       handler(req, res, params)
     })
+    return
   }
+  route = method.toUpperCase() + '/' + route.replace(/^[#/]/, '')
+  this._router.on(route, function (params, req, res) {
+    handler(req, res, params)
+  })
 }
 
 ServerRouter.prototype.match = function (req, res) {

--- a/package.json
+++ b/package.json
@@ -4,7 +4,7 @@
   "description": "Server router",
   "main": "index.js",
   "scripts": {
-    "deps": "dependency-check . && dependency-check . --extra --no-dev",
+    "deps": "dependency-check . --extra --no-dev",
     "test": "standard && npm run deps && NODE_ENV=test node test",
     "test:cov": "standard && npm run deps && NODE_ENV=test istanbul cover test.js"
   },
@@ -17,16 +17,16 @@
   ],
   "license": "MIT",
   "dependencies": {
-    "pathname-match": "^1.1.3",
+    "pathname-match": "^1.2.0",
     "urlencode": "^1.1.0",
-    "wayfarer": "^6.1.4"
+    "wayfarer": "^7.0.0"
   },
   "devDependencies": {
-    "concat-stream": "^1.5.1",
-    "dependency-check": "^2.5.1",
+    "concat-stream": "^2.0.0",
+    "dependency-check": "^4.1.0",
     "get-server-port": "^1.0.0",
-    "istanbul": "^0.4.1",
-    "standard": "^8.0.0",
-    "tape": "^4.2.2"
+    "istanbul": "^0.4.5",
+    "standard": "^14.3.1",
+    "tape": "^4.11.0"
   }
 }

--- a/test.js
+++ b/test.js
@@ -43,7 +43,7 @@ tape('server-router', function (t) {
     t.plan(1)
 
     var router = serverRouter({ default: '/foo' })
-    router.route('GET', '/foo', function (req, res, ctx) {
+    router.route('', '/foo', function (req, res, ctx) {
       t.pass('/foo called')
       res.end()
       server.close()
@@ -61,8 +61,8 @@ tape('server-router', function (t) {
     t.plan(1)
 
     var router = serverRouter({ default: '/foo' })
-    router.route('GET', '/foo/:bar', function (req, res, ctx) {
-      t.equal(ctx.params.bar, 'hi-there')
+    router.route('GET', '/foo/:bar', function (req, res, params) {
+      t.equal(params.bar, 'hi-there')
       res.end()
       server.close()
     })

--- a/walk.js
+++ b/walk.js
@@ -4,8 +4,8 @@ var assert = require('assert')
 module.exports = walkServerRouter
 
 function walkServerRouter (router, cb) {
-  assert.equal(typeof router, 'function', 'server-router/walk: router should be a function')
-  assert.equal(typeof cb, 'function', 'server-router/walk: cb should be a function')
+  assert.strictEqual(typeof router, 'function', 'server-router/walk: router should be a function')
+  assert.strictEqual(typeof cb, 'function', 'server-router/walk: cb should be a function')
 
   router = router._router
   return walk(router, cb)

